android/tangram/src/com/mapzen/tangram/TouchLabel.java

```diff
@@ -0,0 +1,47 @@
+package com.mapzen.tangram;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * {@code TouchLabel} represents labels that can be selected on the screen
+ */
+public class TouchLabel {
+
+    /**
+     * Options for the type of TouchLabel
+     */
+    public enum LabelType {
+        ICON,
+        TEXT,
+    }
+
+    private LngLat coordinate;
+    private LabelType type;
+    private Map<String, String> properties;
+
+    public TouchLabel(double[] coordinates, int type, Map<String, String> properties) {
```

Comment:
Recommend making this package-private, I don't think SDK users should be creating these objects manually.